exo/inference/mlx/sharded_inference_engine.py

```diff
@@ -53,24 +69,84 @@ async def decode(self, shard: Shard, tokens) -> str:
     await self.ensure_shard(shard)
     tokens = await asyncio.get_running_loop().run_in_executor(self.executor, self.tokenizer.decode, tokens)
     return tokens
+
+  async def save_checkpoint(self, shard: Shard, path: str):
+    await self.ensure_shard(shard)
+    await asyncio.get_running_loop().run_in_executor(self.executor, self.model.save_weights, path)
+
+  async def load_checkpoint(self, shard: Shard, path: str):
+    await self.ensure_shard(shard)
+    await asyncio.get_running_loop().run_in_executor(self.executor, self.model.load_weights, path)
     
   async def infer_tensor(self, request_id: str, shard: Shard, input_data: np.ndarray) -> np.ndarray:
     await self.ensure_shard(shard)
-    output_data: np.ndarray = np.array(await asyncio.get_running_loop().run_in_executor(self.executor, self.model, mx.array(input_data), request_id))
+    loop = asyncio.get_running_loop()
+    state = await self.poll_state(request_id)
+    x = mx.array(input_data).astype(mx.int64) if self.shard.is_first_layer() else mx.array(input_data)
```

Comment:
what does this line do? what's the cast to int64 for?

Reply:
I pulled that from an MLX example because at some point the inference engine didn't like the way the tokens were encoded, but I think it's unnecessary now as they're properly typed elsewhere in the code. Will test

Reply:
I guess it worked since you removed it!